config/rbac/cryostat_role.yaml

```diff
@@ -12,8 +13,49 @@ rules:
   verbs:
   - create
 - apiGroups:
-  - authorization.k8s.io
+  - ""
   resources:
-  - selfsubjectaccessreviews
+  - pods
   verbs:
   - create
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - replicationcontrollers
+  - endpoints
+  verbs:
+  - get
+- apiGroups:
+  - operator.cryostat.io
+  resources:
+  - cryostats
+  verbs:
+  - create
+  - delete
+  - get
+- apiGroups:
+  - operator.cryostat.io
+  resources:
+  - flightrecorders
+  - recordings
+  verbs:
+  - create
+  - delete
+  - get
+  - patch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - create
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  - replicasets
+  - statefulsets
+  verbs:
+  - get
```

Comment:
Setting this PR to draft mode while I add the new ClusterRole

Reply:
I've removed `create TokenReviews` and `create SelfSubjectAccessReviews` from the new ClusterRole. The backend can request a token with enough permissions using `user:info user:check-access role:cryostat-operator-oauth-client:namespace`.

Reply:
Hey Janelle, I haven't tried it out yet, but do you know if the `user:info` scope allows the backend to create a TokenReview, or does that code need to be changed to use the OpenShift users API? I assumed it was the latter.

Reply:
I'm assuming the `user:info` scope lets the backend create a TokenReview, because when I tested my latest image on OpenShift with the following ClusterRole permissions and `scope=user%3Ainfo+user%3Acheck-access+role%3Acryostat-operator-oauth-client` as the role scope, I didn't see any errors on the frontend or in the `cryostat-sample` logs.

```
Name:         cryostat-operator-oauth-client
Labels:       <none>
Annotations:  <none>
PolicyRule:
  Resources                             Non-Resource URLs  Resource Names  Verbs
  ---------                             -----------------  --------------  -----
  flightrecorders.operator.cryostat.io  []                 []              [create delete get patch]
  recordings.operator.cryostat.io       []                 []              [create delete get patch]
  cryostats.operator.cryostat.io        []                 []              [create delete get]
  pods                                  []                 []              [create get]
  deployments.apps                      []                 []              [create get]
  endpoints                             []                 []              [get]
  replicationcontrollers                []                 []              [get]
  daemonsets.apps                       []                 []              [get]
  replicasets.apps                      []                 []              [get]
  statefulsets.apps                     []                 []              [get]
```

Reply:
I think what we're seeing is the TokenReview is being done using Cryostat's ServiceAccount: https://github.com/cryostatio/cryostat/blob/7c010b29b83677ba0a815315ddd50ada6afb5659/src/main/java/io/cryostat/net/OpenShiftAuthManager.java#L270-L288

It still has the `create tokenreviews` permission from the existing ClusterRole. So the `user:info` scope is not being used right now. Later on, if we change the TokenReview to an OpenShift user API query using the scoped token, the `user:info` scope will be necessary. I think that's probably best saved for another PR though.

Reply:
Thanks for the clarification. I'll remove `user:info` from the backend request then.